libretroshare/src/rsserver/p3peers.cc

```diff
@@ -248,6 +248,9 @@ bool    p3Peers::isFriend(const RsPeerId &ssl_id)
         return mPeerMgr->isFriend(ssl_id);
 }
 
+bool p3Peers::isFriendPendingPgp(const RsPeerId& sslId)
+{ return isFriend(sslId) && !isPgpFriend(getGPGId(sslId)); }
```

Comment:
/!\ Why dont you use a flag in the Peer structure to mark the SSL id as not yet validated using PGP? That would prevent the risk of calling isPgpFriend on a Pgp id that is actually not even yet known to be the one that will sign the handshake.
Typical attack vector: I send you a certificate with (SSL Id, PGP if of a friend of yours). This method will class the short invite as a true friend without any checking first.
The safe way to do this is to base the "friendship level" on a flag that is set when you add the friend from short invite and reset after the PGP key is received and actually validates the certificate.

Reply:
No it does not. IsPGPFriend only looks for the PGP fingerprint in a list of accepted peers, but you certainly cannot use that function to allow access to a SSL certificate the signature of which youhave not tested yet. Here you are doing a typical mistake: use a trick that makes coding shorter but at the expense of security.

Reply:
Well it is what I expect from `isPgpFriend(PgpId)` just check in the list of friend PGP ids and return false if it is not a friend. We cannot verify the PGP signature there because we don't have the PGP key as expected. Perhaps `isFriendPendingPgp(RsPeerId)` isn't a good name... What this method is supposed to do is to check if an RsPeerId is an SSL-only friend, so it return false for PGP friends, and for non-friends, while it return true only for SSL-only friends. Is `isSslOnlyFriend(RsPeerId)` more understandable?